File: src/dataset/javascript/acronym_exercism-javascript-ThomasZumsteg/openAI/ChatGPT4_acronym_v3.js
export default {
    parse: parse
};

function parse(phrase) {
    return phrase.split(/\W+/)
        .map(word => {
            const firstChar = word.charAt(0).toUpperCase();
            return /^[A-Z]+$/.test(word) 
                ? firstChar 
                : firstChar + [...word.slice(1)].filter(char => char >= 'A' && char <= 'Z').join('');
        })
        .join('');
}

File: src/dataset/Java/acronym_exercism-java-uzilan_robiworks/openAI/ChatGPT4_acronym_v3.java
class Acronym {
    private final String phrase;

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
        StringBuilder abbreviation = new StringBuilder();
        for (String word : phrase.split("[^a-zA-Z']+")) {
            if (!word.isEmpty()) {
                abbreviation.append(Character.toUpperCase(word.charAt(0)));
            }
        }
        return abbreviation.toString();
    }
}